vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.ts

```diff
@@ -700,128 +475,38 @@ window.Vaadin.Flow.gridConnector.initLazy = (grid) => {
     let updatedPageCount = Math.ceil(length / grid.pageSize);
 
     // For root cache, remember the range of pages that were cleared during an update
-    if (pkey === root) {
-      currentUpdateClearRange = [firstPage, firstPage + updatedPageCount - 1];
-    }
+    currentUpdateClearRange = [firstPage, firstPage + updatedPageCount - 1];
 
     for (let i = 0; i < updatedPageCount; i++) {
       let page = firstPage + i;
-      let items = cache[pkey][page];
-      grid.$connector.doDeselection(items.filter((item) => selectedKeys[item.key]));
-      items.forEach((item) => grid.closeItemDetails(item));
-      delete cache[pkey][page];
-      updateGridCache(page, parentKey);
-      updateGridItemsInDomBasedOnCache(items);
```

Comment:
There is already a call to `grid.__updateVisibleRows()` below which triggers a re-render.